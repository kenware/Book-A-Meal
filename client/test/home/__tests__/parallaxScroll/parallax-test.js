import React from 'react';
import { shallow } from 'enzyme';
import ParallaxScrol from '../../../../components/home/parallaxScroll';

describe('Test parallax components', () => {
  it('renders correctly', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree).toMatchSnapshot();
  });

  it('should have one parallax container', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('.parallex')).toHaveLength(1);
  });

  it('should have three parallax-menu container', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('.parallex-menu').length).toEqual(3);
  });

  it('should have three parallex-img container', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('.parallex-img').length).toEqual(3);
  });

  it('should have 16 div', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('div').length).toEqual(16);
  });

  it('should have three link', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('Link').length).toEqual(3);
  });

  it('it should have three container class', () => {
    const tree = shallow(<ParallaxScrol />);
    expect(tree.find('.container')).toHaveLength(3);
  });
});
