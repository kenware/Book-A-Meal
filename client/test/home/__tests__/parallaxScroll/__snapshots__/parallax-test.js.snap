// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test parallax components renders correctly 1`] = `
<div
  className="parallex"
>
  <div
    className="parallex-menu"
  >
    <div
      className="parallex-img"
    >
      <span
        className="fa fa-dashboard fa-4x white-color"
        style={
          Object {
            "marginLeft": "40%",
          }
        }
      />
    </div>
    <div
      className="parallex-card"
    >
      <div
        className="container"
      >
        <br />
        <h2
          className="white-color"
        >
          Dashboard
        </h2>
        <p
          className="white-color"
        >
          Enter your dashboard to perform awesome functionality which include ordering for a meal, adding meal options from meal menu
        </p>
      </div>
      <div
        className="parallex-footer"
      >
        <button>
          <Link
            replace={false}
            to="/dashbord"
          >
             more
          </Link>
        </button>
      </div>
    </div>
  </div>
  <div
    className="parallex-menu"
  >
    <div
      className="parallex-img"
    >
      <span
        className="fa fa-rocket fa-4x white-color"
        style={
          Object {
            "marginLeft": "40%",
          }
        }
      />
    </div>
    <div
      className="parallex-card"
    >
      <div
        className="container"
      >
        <br />
        <h2
          className="white-color"
        >
          Login/SignUp
        </h2>
        <p
          className="p-style"
        >
          Login to order menu of your choice with just a single click
        </p>
      </div>
      <div
        className="parallex-footer"
      >
        <button>
          <Link
            replace={false}
            to="/login"
          >
             more
          </Link>
        </button>
      </div>
    </div>
  </div>
  <div
    className="parallex-menu"
  >
    <div
      className="parallex-img"
    >
      <span
        className="fa fa-user fa-4x white-color"
        style={
          Object {
            "marginLeft": "40%",
          }
        }
      />
    </div>
    <div
      className="parallex-card"
    >
      <div
        className="container"
      >
        <br />
        <h2
          className="white-color"
        >
          SignUp
        </h2>
        <p
          className="p-style"
        >
          SignUp to get started. Login to order menu of your choice with just a single click
        </p>
      </div>
      <div
        className="parallex-footer"
      >
        <button>
          <Link
            replace={false}
            to="/register"
          >
             more
          </Link>
        </button>
      </div>
    </div>
  </div>
</div>
`;
